test/components/kubernetes/baremetal_kubelet_labels_test.go

```diff
@@ -0,0 +1,75 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build baremetal
+// +build e2e
+
+package kubernetes //nolint:testpackage
+
+import (
+	"context"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkglabels "k8s.io/apimachinery/pkg/labels"
+
+	testutil "github.com/kinvolk/lokomotive/test/components/util"
+)
+
+func TestBaremetalNodeHasNodeSpecificLabels(t *testing.T) {
```

Comment:
I think we could name this test `Test_Baremetal_NodeSpecificLabels` and then has 2 sub-tests for it, one for checking that labels has been applied, and other one to check that common labels are retained.

Reply:
I'd rather append the labels provided in ci configuration in
```
    nodeNamesAndExpectedLabelsInCI := map[string]map[string]string{
      "mercury-controller-0": {
        "testkey": "testvalue",
        "node.kubernetes.io/master": "",
        "node.kubernetes.io/controller": "true",
      },
      "mercury-worker-0": {
        "ingressnode": "yes",     
        "testing.io": "yes",
        "roleofnode": "testing",
      },
      "mercury-worker-1": {
        "storagenode": "yes",
        "testing.io": "yes",
        "roleofnode": "testing"
      },
    }
    ```
    
and pass map of specific node to LabelSelector, which would confirm that node specifc labels were applied as well as  confirm that common labels were retained.

Reply:
Sounds good.